testing/util/bellatrix.go

```diff
@@ -53,3 +63,176 @@ func BlockSignatureBellatrix(
+// Use BlockGenConfig to declare the conditions you would like the block generated under.
+// This function modifies the passed state as follows:
+
+func GenerateFullBlockBellatrix(
+	bState state.BeaconState,
+	privs []bls.SecretKey,
+	conf *BlockGenConfig,
+	slot types.Slot,
+) (*ethpb.SignedBeaconBlockBellatrix, error) {
+	ctx := context.Background()
+	currentSlot := bState.Slot()
+	if currentSlot > slot {
+		return nil, fmt.Errorf("current slot in state is larger than given slot. %d > %d", currentSlot, slot)
+	}
+	bState = bState.Copy()
+
+	if conf == nil {
+		conf = &BlockGenConfig{}
+	}
+
+	var err error
+	var pSlashings []*ethpb.ProposerSlashing
+	numToGen := conf.NumProposerSlashings
+	if numToGen > 0 {
+		pSlashings, err = generateProposerSlashings(bState, privs, numToGen)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed generating %d proposer slashings:", numToGen)
+		}
+	}
+
+	numToGen = conf.NumAttesterSlashings
+	var aSlashings []*ethpb.AttesterSlashing
+	if numToGen > 0 {
+		aSlashings, err = generateAttesterSlashings(bState, privs, numToGen)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed generating %d attester slashings:", numToGen)
+		}
+	}
+
+	numToGen = conf.NumAttestations
+	var atts []*ethpb.Attestation
+	if numToGen > 0 {
+		atts, err = GenerateAttestations(bState, privs, numToGen, slot, false)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed generating %d attestations:", numToGen)
+		}
+	}
+
+	numToGen = conf.NumDeposits
+	var newDeposits []*ethpb.Deposit
+	eth1Data := bState.Eth1Data()
+	if numToGen > 0 {
+		newDeposits, eth1Data, err = generateDepositsAndEth1Data(bState, numToGen)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed generating %d deposits:", numToGen)
+		}
+	}
+
+	numToGen = conf.NumVoluntaryExits
+	var exits []*ethpb.SignedVoluntaryExit
+	if numToGen > 0 {
+		exits, err = generateVoluntaryExits(bState, privs, numToGen)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed generating %d attester slashings:", numToGen)
+		}
+	}
+
+	numToGen = conf.NumTransactions
+	newTransactions := make([][]byte, numToGen)
+	for i := uint64(0); i < numToGen; i++ {
+		newTransactions[i] = bytesutil.Uint64ToBytesLittleEndian(i)
+	}
+	random, err := helpers.RandaoMix(bState, time.CurrentEpoch(bState))
+	if err != nil {
+		return nil, errors.Wrap(err, "could not process randao mix")
+	}
+
+	timestamp, err := slots.ToTime(bState.GenesisTime(), slot)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get current timestamp")
+	}
+
+	stCopy := bState.Copy()
+	stCopy, err = transition.ProcessSlots(context.Background(), stCopy, slot)
+	if err != nil {
+		return nil, err
+	}
+
+	parentExecution, err := stCopy.LatestExecutionPayloadHeader()
+	if err != nil {
+		return nil, err
+	}
+	blockHash := indexToHash(uint64(slot))
+	newExecutionPayload := &enginev1.ExecutionPayload{
```

Comment:
This can be improved by including transactions and configuring some of the fields that are hardcoded here.